test/Libraries/PackageManagerTests/PackageLoaderTests.cs

```diff
@@ -107,6 +107,91 @@ public void PackageLoaderRequestsExtensionsBeLoaded()
             Assert.IsTrue(packageLoaded);
         }
 
+        [Test]
+        public void PackageDoesNotReloadOnAbsenceOfNewPackagePath()
+        {
+            var pathManager = new Mock<Dynamo.Interfaces.IPathManager>();
+            pathManager.SetupGet(x => x.PackagesDirectories).Returns(
+                () => new List<string> { PackagesDirectory });
+
+            var loader = new PackageLoader(pathManager.Object);
+            var libraryLoader = new ExtensionLibraryLoader(CurrentDynamoModel);
+
+            loader.PackagesLoaded += libraryLoader.LoadPackages;
+            loader.RequestLoadNodeLibrary += libraryLoader.LoadLibraryAndSuppressZTSearchImport;
+
+            var packagesLoaded = false;
+            loader.PackagesLoaded += (x) => { packagesLoaded = true; };
+            CurrentDynamoModel.PreferenceSettings.CustomPackageFolders = new List<string>();
+            var loadPackageParams = new LoadPackageParams
+            {
+                Preferences = CurrentDynamoModel.PreferenceSettings,
+                
+            };
+            loader.LoadAll(loadPackageParams);
+            Assert.AreEqual(18, loader.LocalPackages.Count());
+            Assert.AreEqual(true, packagesLoaded);
+
+            var entries = CurrentDynamoModel.SearchModel.SearchEntries.ToList();
+            Assert.IsTrue(entries.Count(x => x.FullName == "Package.Package.Package.Hello") == 1);
+
+            packagesLoaded = false;
+            loadPackageParams.NewPaths = new List<string>();
+            // This function is called upon addition of new package paths in the UI.
+            loader.LoadCustomNodesAndPackages(loadPackageParams, CurrentDynamoModel.CustomNodeManager);
+            Assert.AreEqual(18, loader.LocalPackages.Count());
+
+            // Assert packages are not reloaded if there are no new package paths.
+            Assert.False(packagesLoaded);
+
+            // Assert there are no duplication of nodes after trying to reload packages.
+            Assert.IsTrue(entries.Count(x => x.FullName == "Package.Package.Package.Hello") == 1);
+
+        }
+
+        [Test]
+        public void NoPackageNodeDuplicatesOnAddingNewPackagePath()
+        {
+            var pathManager = new Mock<Dynamo.Interfaces.IPathManager>();
+            pathManager.SetupGet(x => x.PackagesDirectories).Returns(
+                () => new List<string> { PackagesDirectory });
+
+            var loader = new PackageLoader(pathManager.Object);
+            var libraryLoader = new ExtensionLibraryLoader(CurrentDynamoModel);
+
+            loader.PackagesLoaded += libraryLoader.LoadPackages;
```

Comment:
I don't think these will cause issues - but I've been surprised before and it's hard to predict when this could cause a memory leak during testing - if it's easy to do can we either
1. check this branch on self serve on master-15
or 
2. unsubscribe at the end of the test?

Reply:
@mjkkirschner I've unsubscribed at the end of all `packageloader `tests including existing tests.